content/spin/javascript-components.md

```diff
@@ -0,0 +1,281 @@
+title = "Building Spin Components in JavaScript"
+template = "spin_main"
+date = "2022-03-14T00:22:56Z"
+enable_shortcodes = true
+---
+
+With JavaScript being a very popular language, Spin provides support for building components with it using the experimental SDK. The development of the Javascript SDK is continually being worked on to improve user experience and add features. 
+
+> This guide assumes you are familiar with the Javascript programming language,
+> but if you are just getting started, be sure to check [the MDN guide](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide).
+
+> All examples from this page can be found in [the Spin repository on GitHub](https://github.com/fermyon/spin-js-sdk/tree/main/examples).
+
+In order to compile JavaScript programs to Spin components, you also need to install a spin plugin `js2wasm` using the following command:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin plugin update
+$ spin plugin install js2wasm
+```
+
+## Installing Templates
+
+The JavaScript/TypeScript templates can be installed from [spin-js-sdk repository](https://github.com/fermyon/spin-js-sdk/tree/main/) using the following command:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin templates install --git https://github.com/fermyon/spin-js-sdk
+```
+
+which will install the `http-js` and `http-ts` templates.
+
+<!-- @nocpy -->
+
+```text
+Copying remote template source
+Installing template http-ts...
+Installing template http-js...
+Installed 2 template(s)
+
++-------------------------------------------------+
+| Name      Description                           |
++=================================================+
+| http-js   HTTP request handler using Javascript |
+| http-ts   HTTP request handler using Typescript |
++-------------------------------------------------+
+```
+
+## Structure of a JS/TS Component
+
+A new JS/TS component can be created using the following command:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin new http-ts hello-world --accept-defaults
+```
+
+This creates a directory of the following structure:
+
+<!-- @nocpy -->
+
+```text
+hello-world/
+├── package.json
+├── package-lock.json
+├── README.md
+├── spin.toml
+├── src
+│   └── index.ts
+├── tsconfig.json
+└── webpack.config.js
+```
+
+The source for the component is present in `src/index.ts`. `Webpack` is used to bundle the component into a single `js` file which will then be compiled to a `wasm` module using the `js2wasm` plugin.
+
+## Building and Running the Template
+
+First, the dependencies for the template need to be installed and then bundled into a single JavaScript file using the following commands:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ cd hello-world
+$ npm install
+$ spin build
```

Comment:
I think the blog post uses `nom run build`, which calls `spin build` — I don't have a preference towards one or the other, but I think we should be consistent here?